test/subdomainregistrar/TestForeverSubdomainRegistrar.js

```diff
@@ -0,0 +1,331 @@
+        .withArgs(node, FixedPricer.address, account, true)
+    })
+  })
+
+  describe('register', () => {
+    let parentExpiry
+
+    beforeEach(async () => {
+      await BaseRegistrar.register(labelhash('test'), account, parentDuration)
+      await BaseRegistrar.setApprovalForAll(NameWrapper.address, true)
+      await NameWrapper.wrapETH2LD(
+        'test',
+        account,
+        CANNOT_UNWRAP,
+        EMPTY_ADDRESS,
+      )
+      ;[, , parentExpiry] = await NameWrapper.getData(node)
+      expect(await NameWrapper.ownerOf(node)).to.equal(account)
+      await SubdomainRegistrar.setupDomain(
+        node,
+        FixedPricerERC20.address,
+        account,
+        true,
+      )
+      await NameWrapper.setApprovalForAll(SubdomainRegistrar.address, true)
+    })
+    it('should allow subdomains to be created', async () => {
+      const balanceBefore = await Erc20WithAccount2.balanceOf(account2)
+
+      const [, fee] = await FixedPricerERC20.price(
+        namehash('test.eth'),
+        'subname',
+        0,
+      )
+
+      await Erc20WithAccount2.approve(
+        SubdomainRegistrar.address,
+        ethers.constants.MaxUint256,
+      )
+
+      await SubdomainRegistrar2.register(
+        node,
+        'subname',
+        account2,
+        EMPTY_ADDRESS,
+        0,
+        [],
+      )
+
+      const balanceAfter = await Erc20WithAccount2.balanceOf(account2)
+      expect(balanceBefore.sub(balanceAfter)).to.equal(fee)
+      const [owner, fuses, expiry] = await NameWrapper.getData(subNode)
+
+      expect(owner).to.equal(account2)
+      expect(expiry).to.equal(parentExpiry - GRACE_PERIOD)
+      expect(fuses).to.equal(CAN_EXTEND_EXPIRY | PARENT_CANNOT_CONTROL)
+    })
+
+    it('should not allow subdomains to be registerd over another domain', async () => {
+      const balanceBefore = await Erc20WithAccount2.balanceOf(account2)
+
+      const [, fee] = await FixedPricerERC20.price(
+        namehash('test.eth'),
+        'subname',
+        0,
+      )
+
+      await Erc20WithAccount2.approve(
+        SubdomainRegistrar.address,
+        ethers.constants.MaxUint256,
+      )
+
+      await SubdomainRegistrar2.register(
+        node,
+        'subname',
+        account2,
+        EMPTY_ADDRESS,
+        0,
+        [],
+      )
+      const balanceAfter = await Erc20WithAccount2.balanceOf(account2)
+      expect(balanceBefore.sub(balanceAfter)).to.equal(fee)
+      const [owner, fuses, expiry] = await NameWrapper.getData(subNode)
+
+      expect(owner).to.equal(account2)
+      expect(expiry).to.equal(parentExpiry - GRACE_PERIOD)
+      expect(fuses).to.equal(CAN_EXTEND_EXPIRY | PARENT_CANNOT_CONTROL)
+
+      await expect(
+        SubdomainRegistrar2.register(
+          node,
+          'subname',
+          account2,
+          EMPTY_ADDRESS,
+          0,
+          [],
+        ),
+      ).to.be.revertedWith(`Unavailable()`)
+    })
+
+    it('Names can extend their own expiry', async () => {
```

Comment:
Is this not possible for RenstalSubdomainRegistrar? If not, there should be a test for that. Otherwise, I am not sure if this test is necessary as it's not extending through subname registrar but rather directly interacting with the name wrapper.